pkg/prometheus/promcfg.go

```diff
@@ -398,28 +398,36 @@ func (cg *configGenerator) generateServiceMonitorConfig(version semver.Version,
 
 	// Relabel namespace and pod and service labels into proper labels.
 	relabelings = append(relabelings, []yaml.MapSlice{
-		{
-			{Key: "source_labels", Value: []string{"__meta_kubernetes_namespace"}},
-			{Key: "target_label", Value: "namespace"},
-		},
-		{
+		{ // Relabel node labels for pre v2.3 meta labels
 			{Key: "source_labels", Value: []string{"__meta_kubernetes_endpoint_address_target_kind", "__meta_kubernetes_endpoint_address_target_name"}},
 			{Key: "separator", Value: ";"},
 			{Key: "regex", Value: "Node;(.*)"},
 			{Key: "replacement", Value: "${1}"},
 			{Key: "target_label", Value: "node"},
 		},
-		{
+		{ // Relabel pod labels for pre v2.3 meta labels
 			{Key: "source_labels", Value: []string{"__meta_kubernetes_endpoint_address_target_kind", "__meta_kubernetes_endpoint_address_target_name"}},
 			{Key: "separator", Value: ";"},
 			{Key: "regex", Value: "Pod;(.*)"},
 			{Key: "replacement", Value: "${1}"},
 			{Key: "target_label", Value: "pod"},
 		},
+		{
+			{Key: "source_labels", Value: []string{"__meta_kubernetes_namespace"}},
+			{Key: "target_label", Value: "namespace"},
+		},
 		{
 			{Key: "source_labels", Value: []string{"__meta_kubernetes_service_name"}},
 			{Key: "target_label", Value: "service"},
 		},
+		{
+			{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_name"}},
+			{Key: "target_label", Value: "pod"},
+		},
+		{
+			{Key: "source_labels", Value: []string{"__meta_kubernetes_pod_node_name"}},
```

Comment:
this meta label does not exist, just delete this entry

Reply:
![image](https://user-images.githubusercontent.com/220643/48059045-65072800-e1b8-11e8-90d0-7fac958ffd8d.png)

I see this on all my pods. That is why I though that having these two meta metrics added to pod and node respectively would be a safer solution than combining the `__meta_kubernetes_endpoint_address_target_kind` and  `__meta_kubernetes_endpoint_address_target_name` metrics.

Reply:
Right, but the `node` is actually meta information that is not identifying for the target, our stance on that is that any additional meta information must be joined onto the target at query time.

Reply:
I would say that `__meta_kubernetes_POD_node_name` is identifying for the target, as the target is the pod. But I understand that it might change much on cluster rebalances etc, creating many more time-series.
Not sure how bad it will be though, as the "previous" time series will go away and a new appear.

Reply:
I'm open to re-discussing the decision, but I'd rather do that in a separate PR from a bug fix. Can we remove it for now from this PR?